lib/instrumentation/span.js

```diff
@@ -63,13 +64,16 @@ Span.prototype.customStackTrace = function (stackObj) {
   this._recordStackTrace(stackObj)
 }
 
-Span.prototype.end = function (endTime) {
+Span.prototype.end = function (endTime, outcome = false) {
```

Comment:
Ah, no, this is slopiness on my part.  A previous version I was working on had added `outcome` and a second, optional parameter.  This is no longer needed -- I'll remove it.

Reply:
Actually -- belay what I said there.  Let me revisit what early week alan was up to.

Reply:
Sorry for the confusion there @trent -- there was additional dead code to remove that clouded the intended design but hadn't caused the tests to fail.  The current code now reflects my current thinking on this.  

Specifically, no, in the end I do not want to redefine the `end` API method of the span object to include an outcome.  It was ultimately too awkward with the `endTime` parameter already being there, but not being a part of most calls to `span.end`  (I'm not a fan of calls like `span.end(null, 'success')`, especially for the common case) 

The `outcome` parameter is not needed here, or in the call to `_setOutcomeFromSpanEnd`, or in the `_setOutcomeFromSpanEnd` method itself.

The internal outcome property is always set from a method -- either `setOutcome` or one of the `_setOutcome...` methods.  

The outcome is frozen when a programmer calls the public `setOutcome` method.  This is to ensure the value set via the public method overrides other values.

The outcome is also frozen when setting  an outcome from an external/exit span ending.  This is to ensure that the explicit setting to a span to _unknown_ in the existing agent code continues to work.  Freezing the span here should not interfere with any other calls to `setOutcome`, as the span immediately ends.

That covers my intent -- happy to talk through any other implications this might have.